README.md

```diff
@@ -37,4 +43,87 @@ Build targets are: net40, net45, netstandard1.4, netstandard2.1, and netcoreapp2
 
 Visual Studio 2019 is required for executing the unit tests. Requirements:
 - NUnit test adaptor (VS Extension)
-- Be sure to set the default processor architecture for tests to x64 (or giant allocation test will fail)
+- Be sure to set the default processor architecture for tests to x64 (or the giant allocation test will fail)
+
+## How It Works
+
+RecyclableMemoryStream improves GC performance by ensuring that the larger buffers used for the streams are put into the gen 2 heap and stay there forever. This should cause full collections to happen less frequently. If you pick buffer sizes above 85K then you will ensure these are placed on the large object heap, which is touched even less frequently by the garbage collector.
+
+The MemoryStreamManager class maintains two separate pools of objects:
+
+1. **Small Pool** - Holds small buffers (of configurable size). Used by default for all normal read/write operations. Multiple small buffers are chained together in the RecyclableMemoryStream class and abstracted into a single stream.
+2. ** Large Pool** - Holds large buffers, which are only used when you must have a single, contiguous buffer, such as a call to GetBuffer(). 
+
+A RecyclableMemoryStream starts out by using a small buffer, chaining additional ones as the stream capacity grows. Should you ever call `GetBuffer()` and the length is greater than single small buffer's capacity, then the small buffers are converted to a single large buffer.
```

Comment:
maybe worth mentioning that you can set a stream size from the start? i.e.

"A RecyclableMemoryStream starts out by using a small buffer, chaining additional ones as the stream capacity grows **(you can also set a start capacity and the stream will have potentially multiple small buffers at the start)**"

your call - this might be too obvious, or perhaps too much detail too soon.

Reply:
not too obvious- this stuff is great